lib/bridge/IrcHandler.js

```diff
@@ -201,6 +207,55 @@ IrcHandler.prototype.onMessage = Promise.coroutine(function*(req, server, fromUs
         return;
     }
 
+    // For topics, set matrix topic as AS bot user
+    if (action.type === "topic") {
+        // Only bridge topics for rooms created by the bridge, via !join or an alias
+        let origins = ["join", "alias"];
+        let mappings = yield this.ircBridge.getStore().getMappingsForChannelByOrigin(
+            server, channel, origins
+        );
+        let promises = mappings.map((entry) => {
+
+            if (this._roomIdToTopic[entry.matrix.getId()] === action.text) {
+                req.log.info(
+                    "Topic of %s already set to '%s'",
+                    entry.matrix.getId(),
+                    action.text
+                );
+                return;
+            }
+
+            req.log.info(
+                "Message is a topic, using AS bot to set topic in room %s: %s",
+                entry.matrix.getId(),
+                action.text
+            );
+
+            return this.ircBridge.getAppServiceBridge().getIntent().setRoomTopic(
+                entry.matrix.getId(),
+                action.text
+            ).then(() => {
+                req.log.info(
+                    "Set topic of %s ('%s', now cached)",
+                    entry.matrix.getId(),
+                    action.text
+                );
+
+                this._roomIdToTopic[entry.matrix.getId()] = action.text;
+            }, (err) => {
+                req.log.error(err.stack);
+            });
+        });
+        if (mappings.length === 0) {
```

Comment:
Whilst logically this is completely valid, from a reader's perspective this feels out of order. You're in effect doing:
```
// pseudocode
mappings = getSomeMappings();
promises = mappings.map(() => { mapToPromise() })
if (mappings.length === 0) {
  log("no mappings!")
}
```

This feels out of order because typically you would **check** `getSomeMappings()` returned something sensible before operating on it, rather than operating on it **then** checking. Like I said, this is valid because it returns an empty array, but it "feels" out-of-order. This is clearer:

```
mappings = getSomeMappings();
if (mappings.length === 0) {
  log("no mappings!")
  return
}
promises = mappings.map(() => { mapToPromise() })

```